service/src/main/java/com/commercetools/pspadapter/payone/PaymentHandler.java

```diff
@@ -51,29 +52,30 @@ public PaymentHandler(String payoneInterfaceName, String tenantName,
      * @return the result of handling the payment
      */
     public PaymentHandleResult handlePayment(@Nonnull final String paymentId) {
-        PaymentHandleResult result = null;
+        int retryCounter = 0;
         try {
-            for (int i = 0; i < RETRIES_LIMIT; i++) {
+            for (; retryCounter < RETRIES_LIMIT; retryCounter++) {
```

Comment:
Isn't the first semicolon unnecessary?

Reply:
it's good you point it out. Unfortunately we need this. In java for loop, we must have a statement/assignment as the first parameter to the for loop. Therefore we cannot remove that semi-colon. We can do it in GoLang 😄